### PR TITLE
chore: Upgrade pydantic, mypy, and others

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

Upgrading pydantic and drf-spectacular so they work together.
Preparing for other PRs, i.e. #19408 

## Changes

- upgrade dependencies, drf-spectacular has gained Pydantic support for example. 
  see https://github.com/tfranzel/drf-spectacular/releases/tag/0.27.0
- unfortunately encountering a mypy issue (due to updated schemas/types or something):  
  https://github.com/python/mypy/issues/15004
  - need mypy>=1.4.0 to fix 😱 
- upgraded mypy
- added [mypy-baseline](https://github.com/orsinium-labs/mypy-baseline) to keep all mypy errors that are coming up as new baseline
  - seems like the easiest solution to gradually work on them
- also: something upgraded some dependency which in turn required syrupy to be updated, which updated all the snapshots 🤪 

## How did you test this code?

- relying on tests

### CODE REVIEW

**Comment by webjunkie** [comment]
@pauldambra I saw that you tried to update mypy before. I need to update it here, because updating pydantic and others breaks it (as noted above). I found mypy-baseline as a tool to work around the issue that the update causes 1229 problems 😅 
Do you think this is a way to go?

File: ee/clickhouse/views/test/__snapshots__/test_clickhouse_experiment_secondary_results.ambr [hunk 1]
```diff
@@ -157,141 +158,5 @@
               prop
      HAVING steps = max_steps)
   GROUP BY prop
-  '
----
-# name: ClickhouseTestExperimentSecondaryResults.test_basic_secondary_metric_results.5
```
Comment (pauldambra) [inline ee/clickhouse/views/test/__snapshots__/test_clickhouse_experiment_secondary_results.ambr:162]: i think this is normally snapshots that no longer exist...

File: ee/clickhouse/views/test/__snapshots__/test_clickhouse_experiments.ambr [hunk 2]
```diff
@@ -286,124 +173,10 @@
               prop
      HAVING steps = max_steps)
   GROUP BY prop
-  '
----
-# name: ClickhouseTestFunnelExperimentResults.test_experiment_flow_with_event_results_and_events_out_of_time_range_timezones.2
```
Comment (pauldambra) [inline ee/clickhouse/views/test/__snapshots__/test_clickhouse_experiments.ambr:291]: same

File: posthog/api/test/__snapshots__/test_annotation.ambr [hunk 3]
```diff
@@ -279,88 +280,10 @@
          AND NOT "posthog_annotation"."deleted")
   ORDER BY "posthog_annotation"."date_marker" DESC
   LIMIT 1000 /*controller='project_annotations-list',route='api/projects/%28%3FP%3Cparent_lookup_team_id%3E%5B%5E/.%5D%2B%29/annotations/%3F%24'*/
-  '
----
-# name: TestAnnotation.test_retrieving_annotation_is_not_n_plus_1.15
```
Comment (pauldambra) [inline posthog/api/test/__snapshots__/test_annotation.ambr:284]: same

File: posthog/api/test/__snapshots__/test_insight.ambr [hunk 4]
```diff
@@ -1538,28 +1539,10 @@
   WHERE (NOT ("posthog_dashboard"."deleted")
          AND "posthog_dashboard"."id" = 2)
   LIMIT 21 /*controller='project_insights-list',route='api/projects/%28%3FP%3Cparent_lookup_team_id%3E%5B%5E/.%5D%2B%29/insights/%3F%24'*/
-  '
----
-# name: TestInsight.test_listing_insights_does_not_nplus1.30
```
Comment (pauldambra) [inline posthog/api/test/__snapshots__/test_insight.ambr:1543]: and again

File: posthog/api/test/dashboards/__snapshots__/test_dashboard.ambr [hunk 5]
```diff
@@ -8254,76 +8237,65 @@
...
   SELECT "posthog_instancesetting"."id",
          "posthog_instancesetting"."key",
          "posthog_instancesetting"."raw_value"
   FROM "posthog_instancesetting"
   WHERE "posthog_instancesetting"."key" = 'constance:posthog:PERSON_ON_EVENTS_ENABLED'
   ORDER BY "posthog_instancesetting"."id" ASC
   LIMIT 1 /*controller='project_dashboards-detail',route='api/projects/%28%3FP%3Cparent_lookup_team_id%3E%5B%5E/.%5D%2B%29/dashboards/%28%3FP%3Cpk%3E%5B%5E/.%5D%2B%29/%3F%24'*/
-  '
----
-# name: TestDashboard.test_loading_individual_dashboard_does_not_prefetch_all_possible_tiles.315
```
Comment (pauldambra) [inline posthog/api/test/dashboards/__snapshots__/test_dashboard.ambr:8314]: same

File: posthog/api/test/notebooks/__snapshots__/test_notebook.ambr [hunk 6]
```diff
@@ -364,66 +357,21 @@
   FROM "posthog_organizationmembership"
   INNER JOIN "posthog_organization" ON ("posthog_organizationmembership"."organization_id" = "posthog_organization"."id")
   WHERE "posthog_organizationmembership"."user_id" = 2 /*controller='project_notebooks-list',route='api/projects/%28%3FP%3Cparent_lookup_team_id%3E%5B%5E/.%5D%2B%29/notebooks/%3F%24'*/
-  '
----
-# name: TestNotebooks.test_updates_notebook.20
```
Comment (pauldambra) [inline posthog/api/test/notebooks/__snapshots__/test_notebook.ambr:369]: same 🫠

**Review by pauldambra — APPROVED** [review]
scanned 59 of 97 files 

some changes to the snapshots (although nothing automatically worrying)

looking at the changelog for syrupy there's

* numerically sort snapshots if possible

which might explain the changes

**Pull request merged**